### PR TITLE
Details on CSV locations and constraints

### PR DESCRIPTION
**Pull request by LotharBender**
- Folder recommendation: _csv_ → _data_
- Data can be put alongside cds files

### CODE REVIEW

File: guides/databases.md [hunk 1]
```diff
@@ -245,14 +245,28 @@ cds add data
 
 ### Location of CSV Files
 
+CSV files can be located in the folders _db/data_ and _test/data_ as well as in any _data_ subfolder of a CDS file if that file is part of the compiled model.
+
+The following conventions apply:
+* Each file contains data for one entity.
+* File names must follow the pattern _namespace-entity.csv_. <br>
+  Pattern for nested entities: _namespace-entity.nestedEntity.csv_. <br>
+  Examples: _my.bookshop-Books.csv_, or _my.bookshop-Books.ISBN.csv_.
```
Comment (chgeo) [inline guides/databases.md:254]: As discussed: remove

File: guides/databases.md [hunk 2]
```diff
@@ -245,14 +245,28 @@ cds add data
 
 ### Location of CSV Files
 
+CSV files can be located in the folders _db/data_ and _test/data_ as well as in any _data_ subfolder of a CDS file if that file is part of the compiled model.
```
Comment (chgeo) [inline guides/databases.md:248]: Detail out `as well as in any _data_ subfolder of a CDS file if that file is part of the compiled model` above in a detail section.

File: guides/databases.md [hunk 3]
```diff
@@ -245,14 +245,28 @@ cds add data
...
 <div markdown="1" class="impl node">
 
-| Location    | Deployed...          | Purpose                                                  |
-| ----------- | -------------------- | -------------------------------------------------------- |
-| `db/data`   | always               | initial data for configurations, code lists, and similar |
-| `test/data` | if not in production | sample data for tests and demos                          |
+| Location          | Deployed...          | Purpose                                                  |
+| ----------------- | -------------------- | -------------------------------------------------------- |
+| `db/data`         | always               | initial data for configurations, code lists, and similar |
+| `db/<*.cds>/data` | always               | initial data for configurations, code lists, and similar |
```
Comment (chgeo) [inline guides/databases.md:268]: Remove again

File: guides/databases.md [hunk 4]
```diff
@@ -245,14 +245,28 @@ cds add data
...
+The following conventions apply:
+* Each file contains data for one entity.
+* File names must follow the pattern _namespace-entity.csv_. <br>
+  Pattern for nested entities: _namespace-entity.nestedEntity.csv_. <br>
+  Examples: _my.bookshop-Books.csv_, or _my.bookshop-Books.ISBN.csv_.
+* They must start with a header line that lists the needed element names.
+
+::: details On SAP HANA ...
+CSV files located in _db/src/**_ folder and corresponding _hdbtabledata_ files will be treated as native SAP HANA artifacts and deployed as they are.
+:::
```
Comment (chgeo) [inline guides/databases.md:259]: Move down

Comment (LotharBender) [reply guides/databases.md]: removed, location info `db/src/**` added to _guides/databases-hana.md_

File: guides/databases-hana.md [hunk 5]
```diff
@@ -534,7 +534,7 @@ The `change-mode` option determines whether `ALTER TABLE ... ALTER` (`"alter"`)
 
 ### CSV Data Gets Overridden
 
-HDI deploys CSV data as _.hdbtabledata_, which assumes exclusive ownership of the data. It's overridden with the next application deployment; hence:
+HDI deploys CSV data as _.hdbtabledata_ located in folder _db/src/**_. HDI assumes exclusive ownership of the data. It's overridden with the next application deployment; hence:
```
Comment (chgeo) [inline guides/databases-hana.md:537]: Mentioning `db/src` location is misleading a bit, as it also true for `gen/db/...`
I would just not mention it here.  The statement here is that any `hdbtabledata` file has this behavior, regardless from where it was deployed (generated, manual, etc).

File: guides/databases.md [hunk 6]
```diff
@@ -245,12 +245,18 @@ cds add data
 
 ### Location of CSV Files
 
+CSV files can be located in the folders _db/data_ and _test/data_ as well as in any _data_ folder next to your CDS model files.
+
+::: details Adding initial data next to your data model
+While the concrete content of the particular CDS model file is irrelevant, it has to be part of your data model. Of course CSV file names must match fully qualified names of respective entity definitions in your CDS models.
```
Comment (chgeo) [inline guides/databases.md:251]:
```suggestion
The content of these 'co-located' `.cds` files actually does not matter, but they need to be included in your data model, through a `using` clause in another file for example.
```

File: guides/databases-hana.md (hunk 5 above)
Comment (chgeo) [inline guides/databases-hana.md:537]:
```suggestion
HDI deploys CSV data as _.hdbtabledata_ and assumes exclusive ownership of the data. It's overridden with the next application deployment; hence:
```

**Review by chgeo — APPROVED** [review]

File: guides/databases.md [hunk 7]
```diff
@@ -242,6 +242,16 @@ cds add data
 
 ### Location of CSV Files
 
+CSV files can be located in the folders _db/data_ and _test/data_ as well as in any _data_ folder next to your CDS model files.
+
+::: details Adding initial data next to your data model
+The content of these 'co-located' `.cds` files actually does not matter, but they need to be included in your data model, through a `using` clause in another file for example.
```
Comment (renejeglinsky) [inline guides/databases.md:248]:
```suggestion
The content of these 'co-located' `.cds` files actually doesn't matter, but they need to be included in your data model, through a `using` clause in another file for example.
```

**Review by renejeglinsky — APPROVED** [review]

**Pull request merged**